addons/mass_mailing/tests/test_mass_mailing.py

```diff
@@ -0,0 +1,124 @@
+        #100% received then the state of mass_mailing should be "Done"
+        self.assertEqual(mass_mailing.state, 'done', "Mass mailing state not set properly")
+        #checking opened Ratio
+        self.MailStats.set_opened(mail_mail_ids=[mass_mailing.statistics_ids[1].mail_mail_id_int])
+        mass_mailing._compute_statistics()
+        self.assertEqual(mass_mailing.opened_ratio, (mass_mailing.opened*100)/mass_mailing.total, "Opened ratio not set properly")
+        #checking replied Ratio
+        self.MailStats.set_replied(mail_mail_ids=[mass_mailing.statistics_ids[2].mail_mail_id_int])
+        mass_mailing._compute_statistics()
+        self.assertEqual(mass_mailing.replied_ratio, (mass_mailing.opened*100)/mass_mailing.total, "Replied ratio not set properly")
+        #checking bounced Ratio
+        self.MailStats.set_bounced(mail_mail_ids=[mass_mailing.statistics_ids[3].mail_mail_id_int])
+        mass_mailing._compute_statistics()
+        self.assertEqual(mass_mailing.bounced_ratio, (mass_mailing.opened*100)/mass_mailing.total, "Bounced ratio not set properly")
+
+    def test_create_mass_mailing_to_mailing_list(self):
+        mailing_list_new = self.MassMailingList.create({
+            'name': "Test mailing List"})
+        mass_mailing = self.MassMailing.create({
+            'name': 'Test Mass Mailing List',
+            'contact_list_ids': [(4, mailing_list_new.id)],
+            'reply_to_mode': 'email',
+            'body_html': "<p>this is a test mass mailing to mailing_list(s)<p>"})
+        mass_mailing.on_change_model_and_list()
+        mass_mailing_contact_01 = self.MassMailingContact.create({
+            'name': 'test',
+            'email': 'test@example.com',
+            'list_id': mailing_list_new.id,
+            })
+        #adding contacts to mailing list
+        mass_mailing_contact_02 = self.MassMailingContact.create({
+            'name': 'test_contact_02',
+            'email': 'test02@example.com',
+            })
+        mass_mailing_contact_03 = self.MassMailingContact.create({
+            'name': 'test_contact_03',
+            'email': 'test03@example.com',
+            })
+        #checking thet wether the contact assigend the newly created mailing list or not-default behaviour
+        self.assertEqual(mass_mailing_contact_03.list_id.id, mailing_list_new.id, "Default list_id not assigned prorly")
+        #adding a new contact to existing mailing list
+        #adding a contact via add_to_list(), actually called from controller
+        self.MassMailingContact.add_to_list("test04@example.com", mailing_list_new.id)
+        contact = self.MassMailingContact.search([('email', '=', "test04@example.com"), ('list_id', '=', mailing_list_new.id)])
+        #checking that newly created contact is properly created or not
+        self.assertEqual(contact.email, "test04@example.com", "add_to_list method not working properly")
```

Comment:
make assert message more functional(understandable) which cannot be contain technical term. better to change message.. so end-user can easily understand..